content/blog/how-to-locate-elements-in-playwright.md

```diff
@@ -0,0 +1,183 @@
+
+```js
+await page.getByRole('link', { name: 'About' }).click();
+```
+
+### Using regex
+
+We can also use a regular expression and ignore the case so that the test will pass regardless of if the text it contains has a capital letter or not.
+
+```js
+await page.getByRole('link', { name: /about/i }).click();
+```
+
+### Setting exact to true
+
+You can set exact to true for an exact match. This is helpful if, for example, you have a link that contains the text "About" and another one that contains the text "About the Company". With exact set to true it will only match the link with "About" and not with "About the Company".
+
+```js
+await page.getByRole('link', { name: 'About', exact: true }).click();
+```
+
+### Adding an aria-label
+
+If a link contains an `aria-label` then the value of the aria label will be used instead of the text content.
+
+```html
+<a href="/locators" aria-label="read more about locators">Read more</a>
+```
+
+We can then locate the link using the [`getByRole()`](https://playwright.dev/docs/locators#locate-by-role) locator of link followed by the value of the aria-label as the name.
+
+```js
+await page.getByRole('link', { name: /read more about locators/i }).click();
+```
+
+When you are writing your tests it can be quite difficult to know which accessible role to use if using the `getByRole()` locator. For a link it is pretty straight forward but there is a vast list of roles that can be used. If your element doesn't have an accessible locator then choosing the best locator to use is not easy especially if you are not familiar with the DOM structure of the site you are testing. So how can you choose the best locator? 
+
+>Don't spend time trying to figure out which is the best locator to use. Instead use the test generator to generate your locator.
+
+---
+
+## Playwright's test generator
+
+You can generate code right from VS Code when using the [VS Code extension](https://playwright.dev/docs/getting-started-vscode#installation) by clicking the 'record new' link from the testing sidebar. 
+
+![test generator in vs code editor](https://res.cloudinary.com/debsobrien/image/upload/v1669817488/debbie.codes/blog/2022/vs-code-codegen_gxf58x.png)
+
+As you click on an element on your website, the generator will look at your DOM and figure out the best locator for you, prioritizing role based locators over other locators such as text or test id locators. If the generator comes across two locators with the same name it will either chain locators or use the [`filter()`](https://playwright.dev/docs/locators#filtering-locators) method to ensure the locator is resilient so you don't have to worry about failing tests due to locators.
```

Comment:
```suggestion
As you click on an element on your website, the generator will look at your page and figure out the best locator for you, prioritizing role, text and test id locators. If the generator finds multiple elements matching the locator, it will improve the locator to make it resilient and uniquely identify the target element, so you don't have to worry about failing tests due to locators. For example, it could chain locators to narrow down the scope, or use the [`filter()`](https://playwright.dev/docs/locators#filtering-locators) method to select between multiple elements.
```